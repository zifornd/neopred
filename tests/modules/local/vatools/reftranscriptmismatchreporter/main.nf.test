nextflow_process {

    name "Test Process VATOOLS_REFTRANSCRIPTMISMATCHREPORTER"
    script "modules/local/vatools/reftranscriptmismatchreporter/main.nf"
    process "VATOOLS_REFTRANSCRIPTMISMATCHREPORTER"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
		input[0] = Channel.of([
                    [ id:'test' ], 
                    [ file("$projectDir/tests/test_result/gunzip/test.annotated.vcf", checkIfExists: true) ]
                ])
		
                """
            }
        }

        then {
            assert process.success
	    assert process.out.filter_vcf
	    assert process.out.versions
            assert snapshot(process.out).match()
        }

    }

}
