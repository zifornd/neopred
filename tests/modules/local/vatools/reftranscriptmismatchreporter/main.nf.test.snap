{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filter.vcf:md5,6c688c54f3125754446271c2dc8a1be5"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8b496a612425850ad3947e69e4b1029b"
                ],
                "filter_vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filter.vcf:md5,6c688c54f3125754446271c2dc8a1be5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8b496a612425850ad3947e69e4b1029b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-08T06:18:38.510468"
    }
}